plugin/storage/es/factory.go

```diff
@@ -278,7 +274,7 @@ func createSpanWriter(
 		if err != nil {
 			return nil, err
 		}
-		if err := writer.CreateTemplates(spanMapping, serviceMapping, cfg.IndexPrefix); err != nil {
+		if err := writer.CreateTemplates(spanMapping, serviceMapping, cfg.Indices.Spans.Prefix); err != nil {
```

Comment:
in fact, if we extract IndexPrefix back to the top level, I don't think we'd need the other idea we discussed on the previous PR of having user-settable "defaults" applicable to each index

Reply:
@yurishkuro are you suggesting that we remove `Prefix` from `IndexOptions` and have it in `Configuration`?

Reply:
Yes (and call it IndexPrefix)

Reply:
btw, I would define it like this
```go
type IndexPrefix string

func (p IndexPrefix) Apply(indexName string) string { /* move logic of applying prefix here, to avoid repeating it */ }
```

Reply:
@yurishkuro couple more questions regarding this: 
- how should we handle setting the `IndexPrefix` in `getMappingTemplateOptions`?
- how should we set the prefix in the span reader and writer? should they be different for the span index vs. service index?

Reply:
1. What specifically? 
2. Before this PR there was just a single CLI flag `index-prefix`, so we never supported different prefix for spans and services.